### PR TITLE
OCM-9240 | test: fixing id:49137,53031

### PR DESCRIPTION
**Pull request by AkashKanni**
[ocm-9240](https://issues.redhat.com//browse/ocm-9240) [CI]Debug and fix: 49137, 53031

$ ginkgo --focus "(53031|49137)" tests/e2e/
Ginkgo detected a version mismatch between the Ginkgo CLI and the version of Ginkgo imported by your packages:
  Ginkgo CLI Version:
    2.11.0
  Mismatched package versions found:
    2.17.1 used by e2e

  Ginkgo will continue to attempt to run but you may see errors (including flag
  parsing errors) and should either update your go.mod or your version of the
  Ginkgo CLI to match.

  To install the matching version of the CLI run
    go install github.com/onsi/ginkgo/v2/ginkgo
  from a path that contains a go.mod file.  Alternatively you can use
    go run github.com/onsi/ginkgo/v2/ginkgo
  from a path that contains a go.mod file to invoke the matching version of the
  Ginkgo CLI.

  If you are attempting to test multiple packages that each have a different
  version of the Ginkgo library with a single Ginkgo CLI that is currently
  unsupported.
  
Running Suite: ROSA CLI e2e tests suite - /home/akanni/OCP-Repository/rosa/tests/e2e
====================================================================================
Random Seed: 1719825336

Will run 2 of 127 specs
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS••SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS

Ran 2 of 127 Specs in 68.811 seconds
SUCCESS! -- 2 Passed | 0 Failed | 0 Pending | 125 Skipped
PASS

Ginkgo ran 1 suite in 1m9.848237088s
Test Suite Passed

### CODE REVIEW

**Comment by radtriste** [comment]
@AkashKanni Ids fixed here and in the issue are not the same. Can you check ?

**Comment by AkashKanni** [comment]
@radtriste i have corrected the issue title on jira card

File: tests/e2e/test_rosacli_idp.go [hunk 1]
```diff
@@ -330,6 +330,7 @@ var _ = Describe("Edit IDP",
 					"-y")
 				Expect(err).To(BeNil())
 				textData = rosaClient.Parser.TextData.Input(output).Parse().Tip()
+				Expect(textData).Should(ContainSubstring("Configuring IDP for cluster '%s'", clusterID))
```
Comment (yasun1) [inline tests/e2e/test_rosacli_idp.go:333]: I do not think it is an excellent choice to validate the general message at each checkpoint. We should validate this in one checkpoint, and the others will do the distinguishing validation.

Comment (AkashKanni) [reply tests/e2e/test_rosacli_idp.go:333]: @yasun1  it is neccessary to validate this step `Expect(textData).Should(ContainSubstring("Configuring IDP for cluster '%s'", clusterID))` for each IDP creation, earlier this was missing thats why the CI job failed  

Do you mean use different validation checkpoints (from the 4 used in this PR) for each idp creation scenario?

•```
 [FAILED] [11.749 seconds]
Edit IDP [It] can create/delete the HTPasswd IDPs - [id:49137] [feature-idp, Critical, day2]
/go/src/github.com/openshift/rosa/tests/e2e/test_rosacli_idp.go:296

  [FAILED] Expected
      <string>: INFO: Configuring IDP for cluster '2c52jne75v1ccbhk444k0ubapp9iqsoe'
      INFO: Identity Provider 'htpasswdn1' has been created.
      
  to contain substring
      <string>: To log in to the console, open
  In [It] at: /go/src/github.com/openshift/rosa/tests/e2e/test_rosacli_idp.go:334 @ 06/26/24 20:40:26.725
```

Comment (yasun1) [reply tests/e2e/test_rosacli_idp.go:333]: I apologize; I did not realize they are for kinds of IDPs; it is truly required for validation.

I realized that almost all of the validation `Expectations are the same; how about collecting them into one function?

File: tests/e2e/test_rosacli_idp.go [hunk 2]
```diff
@@ -15,6 +15,14 @@ import (
 	ph "github.com/openshift/rosa/tests/utils/profilehandler"
 )
 
+func validateOuptput(textData string, clusterID string, idpName string) {
```
Comment (yasun1) [inline tests/e2e/test_rosacli_idp.go:18]: As you define the function at the top level. I would suggest using a prominent function name so there is no conflict with the others in the future, e.g., `validateIDPOutput`.

Comment (AkashKanni) [reply tests/e2e/test_rosacli_idp.go]: I have updated the PR and i do like your suggestion for function naming, i will try in incorporate this in future,
could you please have a look again? Thanks!

**Pull request merged**